tools/run-mypy

```diff
@@ -68,9 +73,14 @@ if not python_files and not pyi_files:
     sys.exit(0)
 
 mypy_args: List[str] = []
+if args.use_daemon:
+    mypy_args += ["run", "--"]
+# --no-error-summary comes before "--" for mypy while it comes after it for dmypy
 if args.quiet:
     mypy_args += ["--no-error-summary"]
-mypy_args += ["--", *python_files, *pyi_files]
+if not args.use_daemon:
+    mypy_args += ["--"]
+mypy_args += [*python_files, *pyi_files]
```

Comment:
Just leave this—the second `--` should be there in both cases.

```shell
mypy --no-error-summary -- file.py
dmypy run -- --no-error-summary -- file.py
```

(You can see why if you add a Python file beginning with `-`, like `-foo.py`.)